app/models/nextUpdates/NextUpdatesResponseModel.scala

```diff
@@ -19,10 +19,16 @@ package models.nextUpdates
 import auth.MtdItUser
 import models.incomeSourceDetails.QuarterTypeElection.orderingByTypeName
 import models.incomeSourceDetails.{PropertyDetailsModel, QuarterReportingType, QuarterTypeCalendar, QuarterTypeStandard}
+import models.nextUpdates.ObligationStatus.Fulfilled
 import play.api.libs.json._
 
 import java.time.LocalDate
 
+case class ObligationStatus(code: String, name: String)
+object ObligationStatus {
+  val Open: ObligationStatus = ObligationStatus("O", "Open")
```

Comment:
Would it be possible to use a sealed trait here, rather than holding a bunch of strings? We can then transform from the "O"/"F" representation when reading-in the json and just have that knowledge in that outer layer?

Reply:
please share an example of what you're thinking because I don't think it's possible to avoid the strings used here, thank you

Reply:
sure will

Reply:
Cool. :)